### PR TITLE
Add determine bill licence flags service

### PR DESCRIPTION
**Pull request by Beckyrose200**
https://eaflood.atlassian.net/browse/WATER-4746

This PR contributes to cleaning up the supplementary billing flagging process, specifically focusing on handling licences removed from a bill run.

When a bill run is created, users can remove licences from it. Depending on the type of bill run, the following actions are required:

Annual Bill Run: The licence must be flagged for inclusion in the next supplementary bill run. Supplementary Bill Run: Since the licence was already flagged (which is why it was included in this bill run), the existing flag must be maintained. This logic is managed by the DetermineBillLicenceService.

Previously, flagging logic for licences was handled entirely in legacy code. While the endpoint in the service repository still processes other aspects of licence removal, the flagging logic has been moved to our new ProcessBillingFlagsService. This service ensures flags are correctly determined before the legacy endpoint is called.

### CODE REVIEW

**Comment by Cruikshanks** [comment]
**Prettified!**

**Review by Jozzey — COMMENTED** [review]
Looks good, just the one comment

File: app/services/licences/supplementary/determine-bill-licence-flags.service.js [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+ * @param {string} billLicenceId - The UUID of the bill licence being removed from a bill run
+ *
+ * @returns {object} - An object containing the related licenceId, regionId, start and end date and
+ * licence supplementary billing flags
+ */
+async function go(billLicenceId) {
+  const { licence, bill, licenceId } = await _fetchBillLicence(billLicenceId)
+
+  const { flagForPreSrocSupplementary, flagForSrocSupplementary, flagForTwoPartTariffSupplementary } = _updateFlags(
+    bill,
```
[the remark this answers is not in the document]
Comment (Beckyrose200) [reply app/services/licences/supplementary/determine-bill-licence-flags.service.js]: Thank you :)

**Review by Jozzey — APPROVED** [review]

**Pull request merged**